### DISCUSSION TITLE
0.29.0 release

### DISCUSSION THREAD

**Discussion by orhun** [ratatui/ratatui#1415] [Discussion]
Hey all, I would like to discuss the features we would like to add for the next release & any possible rough edges. @ratatui/maintainers 

**Planned release date:** Monday next week (21st)

# Milestone:

- Issues: https://github.com/ratatui/ratatui/issues?q=sort%3Aupdated-desc+is%3Aissue+milestone%3Av0.29.0+is%3Aopen
- PRs: https://github.com/ratatui/ratatui/pulls?q=sort%3Aupdated-desc+is%3Aissue+milestone%3Av0.29.0+is%3Aopen
- Filtered candidate PRs: https://github.com/ratatui/ratatui/pulls?q=is%3Apr+is%3Aopen+sort%3Acreated-asc+draft%3Afalse+-label%3A%22Status%3A+Abandoned%22

# Todo:

- [x] #1375
Fixes #1378
Almost ready, would be nice to get another review.

- [x] #1414
Should be clarified/fixed soon.

- [x] #1358 
Should be good to go.

- [x] #1341
Should be good to go.

- [x] #936
Seems abandoned - we can close this. (Moved to draft

- [x] #917, #919 & #920
These PRs have overlap with each other and the code is not in the best shape. We need to follow up with a review on #917 or decide what to do with them.

- [x] #1403
Problematic. Needs some thoughts.

- [x] https://github.com/ratatui/ratatui/pull/1398
needs reviews

- [x] https://github.com/ratatui/ratatui/pull/981
needs some CI investigation. I'd (JM) suggest this should be included in 0.29.0 if time permits
Decided to defer to after 0.29.0 due to CI and windows problems (which we fixed a while ago, so needs some investigation)

- [x] https://github.com/ratatui/ratatui/pull/1326
Ready. Requires a small merge resolution in the breaking changes doc and crafting a commit message suitable for the changelog

---

Also, I suggest that we close the abandoned PRs/issues before the release to get a clear view of what to work on after 0.29.0

Did I miss anything? Please let me know! 🙂 

(feel free to edit this post)

**Comment by kdheepak** [comment]
This might be good to go too: https://github.com/ratatui/ratatui/pull/1398 but needs reviews

**Comment by orhun** [comment]
We are pretty close :) I will be writing the highlights and smoothening the rough edges over the weekend.

**Comment by joshka** [comment]
More to add:
- [x] https://github.com/ratatui/ratatui/pull/1437
- [x] https://github.com/ratatui/ratatui/pull/1436
- [x] https://github.com/ratatui/ratatui/pull/1434
- [x] https://github.com/ratatui/ratatui/pull/1429

**Comment by orhun** [comment]
Release is cooked:
- [x] https://github.com/ratatui/ratatui/pull/1444
- [x] https://github.com/ratatui/ratatui-website/pull/752

**Comment by orhun** [comment]
Thank you guys @ratatui/maintainers :)
